mm0-rs/src/util.rs

```diff
@@ -120,15 +127,26 @@ impl fmt::Debug for Span {
 }
 
 lazy_static! {
+  /// A PathBuf created by lazy_static! pointing to a canonicalized "."
   static ref CURRENT_DIR: PathBuf =
     std::fs::canonicalize(".").expect("failed to find current directory");
 }
 
+/// Given a PathBuf 'buf', constructs a relative path from [`CURRENT_DIR`] 
+/// to buf, returning it as a String.
+///
+/// Example : If CURRENT_DIR is /home/johndoe/mm0, and buf is 
+/// /home/johndoe/Documents/ahoy.mm1 will return "../Documents/ahoy.mm1"
+///
+/// [`CURRENT_DIR`]: struct.CURRENT_DIR.html
 fn make_relative(buf: &PathBuf) -> String {
   pathdiff::diff_paths(buf, &CURRENT_DIR).as_ref().unwrap_or(buf)
     .to_str().unwrap().to_owned()
 }
 
+/// Arc 3-tuple of a filepath, relative path from CURRENT_DIR, and [`Url`]
```

Comment:
```suggestion
/// A reference to a file. It wraps an `Arc` so it can be cloned thread-safely.
/// A `FileRef` can be constructed either from a `PathBuf` or a (`file://`) [`Url`],
/// and provides (precomputed) access to these views using `path()` and `url()`,
/// as well as `rel()` to get the relative path from CURRENT_DIR.
```